Add filter to griddata

JacquesBorsenberger opened the pull request.

JacquesBorsenberger committed:
WIP add filters

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	src/components/contingency-list.js

JacquesBorsenberger committed:
WIP add filters

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	src/components/contingency-list.js

JacquesBorsenberger committed:
WIP add filters

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	src/components/contingency-list.js

JacquesBorsenberger committed:
Merge branch 'master' of https://github.com/gridsuite/gridactions-app into add_filter_to_griddata

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package.json
#	src/components/app.js
#	src/redux/reducer.js

JacquesBorsenberger committed:
cleanup

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package.json
#	src/components/app.js
#	src/redux/reducer.js

JacquesBorsenberger committed:
cleanup

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package.json
#	src/components/app.js
#	src/redux/reducer.js

JacquesBorsenberger committed:
cleanup

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package.json
#	src/components/app.js
#	src/redux/reducer.js

JacquesBorsenberger committed:
cleanup

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package.json
#	src/components/app.js
#	src/redux/reducer.js

FranckLecuyer commented on src/utils/rest-api.js:
```diff
@@ -246,3 +248,64 @@ export function addFiltersContingencyList(name, newFiltersContingency) {
+}
+
+/**
+ * Get all contingency lists
```
Change comment

FranckLecuyer commented on src/utils/rest-api.js:
```diff
@@ -246,3 +248,64 @@ export function addFiltersContingencyList(name, newFiltersContingency) {
+}
+
+/**
+ * Delete contingency list by name
```
Change comment

FranckLecuyer commented on src/utils/rest-api.js:
```diff
@@ -246,3 +248,64 @@ export function addFiltersContingencyList(name, newFiltersContingency) {
+}
+
+/**
+ * Get contingency list by type
```
Change comment

FranckLecuyer commented on src/utils/rest-api.js:
```diff
@@ -246,3 +248,64 @@ export function addFiltersContingencyList(name, newFiltersContingency) {
+}
+
+/**
+ * Rename list by name
```
Change comment

FranckLecuyer commented on src/translations/en.json:
```diff
@@ -22,11 +28,14 @@
     "FILTERS": "Filters",
     "TWO_WINDINGS_TRANSFORMER": "2W TRANSFORMER",
     "DANGLING_LINE": "DANGLING LINE",
-    "LINE": "LINE",
```
"LINE" should remain

FranckLecuyer commented on src/translations/fr.json:
```diff
@@ -4,6 +4,11 @@
     "addNewContencyFile": "Nom de la nouvelle liste",
```
Coherence problems between french and english json files
Ex : "Min", "Max"  <---> "MinNominalVoltage", "MaxNominalVoltage"

FranckLecuyer commented on src/components/data-tabs.js:
```diff
@@ -0,0 +1,61 @@
+        {
+            name: 'FILTERS',
+            render: () => (
+                <FilterList FilterRenderer={GenericFilter} listType={'line'} />
```
unused props

FranckLecuyer commented on src/components/contingency-list.js:
```diff
@@ -196,7 +185,7 @@ const CustomTooltip = (props) => {
 };
 
-const ContingencyLists = () => {
+const ContingencyLists = ({ listType, FilterRenderer }) => {
```
Don't know the utility of these props

FranckLecuyer commented on src/components/contingency-list.js:
```diff
@@ -424,7 +413,7 @@ const ContingencyLists = () => {
             .then((response) => {
                 if (response.ok) {
-                    getContingencyLists().then((data) => {
+                    getContingencyLists(listType).then((data) => {
```
getContingencyLists has no argument

FranckLecuyer commented on src/components/filter-list.js:
```diff
@@ -0,0 +1,427 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
2021

FranckLecuyer commented on src/components/data-tabs.js:
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
```
Missing copyright

FranckLecuyer commented on src/components/generic-filter.js:
```diff
@@ -0,0 +1,246 @@
+import { FormattedMessage } from 'react-intl';
```
Missing copyright

FranckLecuyer commented on src/components/filters.js:
```diff
@@ -0,0 +1,163 @@
+import React, { useRef, useState } from 'react';
```
Missing copyright

FranckLecuyer commented on src/components/custom-list-item.js:
```diff
@@ -0,0 +1,142 @@
+import { ScriptTypes } from '../utils/script-types';
```
Missing copyright

JacquesBorsenberger committed:
review

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package.json
#	src/components/app.js
#	src/redux/reducer.js

JacquesBorsenberger committed:
review

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

FranckLecuyer commented:
When we change nominal voltage filter from 'range' or 'approx' to 'equality', the 'value2' field remains unchanged in the sql table 'numericfilter', while it should be deleted

FranckLecuyer commented:
Labels 'equality', 'range', 'approx' are not aligned with the combobox arrow

FranckLecuyer commented on src/translations/fr.json:
```diff
@@ -39,5 +46,16 @@
+    "newFilter": "Nouveau filtre",
+    "FilterName": "Nom",
+    "addNewFilter": "Créer un filtre",
+    "substationName": "Nom Poste",
```
If it is 'substationName, french translation should be 'Nom Site'
If it is 'Nom Poste', the english translation should be 'VoltageLevel Name' and the key renamed

JacquesBorsenberger committed:
remove unwanted console log
review :
reset second field in range selection when equality is selected
correct french translation

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

niconoir commented on src/components/filter-list.js:
```diff
@@ -0,0 +1,427 @@
+            <PopupInfo
+                title={<FormattedMessage id="deleteFilter" />}
+                customAlertMessage={
+                    <FormattedMessage id="alertBeforeDeleteFilter" />
```
Message translation alertBeforeDeleteFilter is not present in translations files (fr.json and en.json)

niconoir commented on src/translations/en.json:
```diff
@@ -39,5 +49,12 @@
+    "substationName": "Substation Name",
+    "renameFilter": "Rename filter",
+    "newFilterName": "new filter name",
+    "deleteFilter": "Delete filter"
```
Missing "alertBeforeDeleteFilter" translation

niconoir commented on src/translations/fr.json:
```diff
@@ -39,5 +46,16 @@
+    "deleteFilter": "supprimer le filtre",
+    "equality": "egalité",
+    "approx": "approximation",
+    "range": "interval"
```
Missing "alertBeforeDeleteFilter" translation

niconoir commented on src/redux/actions.js:
```diff
@@ -36,3 +36,13 @@ export function updateContingencyList(contingencyLists) {
+export const UPDATE_FILTER_LIST = 'UPDATE_FILTER_LIST';
+
+export function updateFilterList(filterList) {
+    console.info(filterList);
```
Is this console logging necessary ?

niconoir commented on src/components/filter-list.js:
```diff
@@ -0,0 +1,427 @@
+        const newFilter = {
+            name: name,
+            type: type === ScriptTypes.SCRIPT ? type : 'LINE',
+            transient: true,
```
Why create the filter as transient ? It makes the creation behaviour different from contingency creation.

niconoir commented on src/components/filter-list.js:
```diff
@@ -0,0 +1,427 @@
+        return (
+            <>
+                <Button
+                    className={classes.addNewList}
```
Here, the button style is quite different from the style used for new list creation in contingency-list. They should be identical, for the contingency list and filter list to look identical (right now, the buttons icons and labels sizes are different).

JacquesBorsenberger replied:
need a whole US to refactor the contingency list

niconoir commented on src/components/filters.js:
```diff
@@ -0,0 +1,171 @@
+        countriesList = require('localized-countries')(
+            require('localized-countries/data/' +
+                navigator.language.substr(0, 2))
+        );
```
The language which is used for countries names localization should be retrieved in the application parameters, it is not necessarily the language of the navigator.

JacquesBorsenberger replied:
done

niconoir commented on src/components/generic-filter.js:
```diff
@@ -0,0 +1,251 @@
+    return (
+        <Grid
+            container
+            direction="row"
```
Why "row" and not "column" ?

JacquesBorsenberger replied:
removed

niconoir commented on src/components/generic-filter.js:
```diff
@@ -0,0 +1,251 @@
+                        >
+                            {[
+                                ...Array.from(Array(definition.occurs).keys()),
+                            ].map((n) => {
```
The problem with this way to generate filters components is that it creates multiple filters with exactly the same label, and thus, we don't exactly know what the filter is used for (i.e. we can have two "countries" filters, and we don't exactly know what they really do)

niconoir reviewed (changes requested):
Contingency list and filter list are implemented in quite different ways, but they look and behave quite the same way. Maybe it would be useful to refactor contingency-list component to make its implementation closer to the filter-list component implementation. This would avoid some bugs like the look of the "New filter" and "New list" button which should be the same but actually look different.

JacquesBorsenberger committed:
Merge remote-tracking branch 'origin/master' into add_filter_to_griddata

# Conflicts:
#	src/components/app.js
#	src/redux/reducer.js

JacquesBorsenberger committed:
save list on creation

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
remarks

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
countries selection in generic filter follow app settings

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
countries selection in generic filter follow app settings

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
unifiy stile between list

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
prettier

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
unify style

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
add number to distinguish sides

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
spelling

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json

JacquesBorsenberger committed:
fix language when browser language is selected

Signed-off-by: jacqbors <jacques.borsenberger@rte-france.com>

# Conflicts:
#	package-lock.json
#	package.json
